enterprise/frontend/src/embedding-sdk/lib/theme/embedding-color-palette.ts

```diff
@@ -13,22 +13,35 @@ import { getEmbeddingChartColors } from "./get-embedding-chart-colors";
  */
 export type MappableSdkColor = Exclude<MetabaseColor, "charts">;
 
+type NEW_SEMANTIC_COLOR =
+  | "text-primary"
+  | "text-secondary"
+  | "text-tertiary"
+  | "text-selected"
+  | "text-brand"
+  | "text-white"
+  | "background"
+  | "background-selected"
+  | "background-disabled"
+  | "background-inverse"
+  | "background-brand";
+
 export const SDK_TO_MAIN_APP_COLORS_MAPPING: Record<
   MappableSdkColor,
-  ColorName
+  (ColorName | NEW_SEMANTIC_COLOR)[]
 > = {
-  brand: "brand",
-  border: "border",
-  filter: "filter",
-  summarize: "summarize",
-  "text-primary": "text-dark",
-  "text-secondary": "text-medium",
-  "text-tertiary": "text-light",
-  background: "bg-white",
-  "background-hover": "bg-light",
-  shadow: "shadow",
-  positive: "success",
-  negative: "danger",
+  brand: ["brand"],
+  border: ["border"],
+  filter: ["filter"],
+  summarize: ["summarize"],
+  "text-primary": ["text-dark", "text-primary"],
```

Comment:
Modify the structure, so some colors could be mapped to more CSS custom properties.